cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/KafkaCruiseControlConfig.java

```diff
@@ -106,19 +107,21 @@
   /**
    * Sanity check for
    * <ul>
-   *   <li>{@link AnalyzerConfig#GOALS_CONFIG} and {@link AnalyzerConfig#INTRA_BROKER_GOALS_CONFIG} are non-empty.</li>
+   *   <li>{@link AnalyzerConfig#GOALS_CONFIG} and {@link AnalyzerConfig#INTRA_BROKER_GOALS_CONFIG} are non-empty and disjoint.</li>
    *   <li>Case insensitive goal names.</li>
    *   <li>{@link AnalyzerConfig#DEFAULT_GOALS_CONFIG} is non-empty.</li>
-   *   <li>{@link AnomalyDetectorConfig#SELF_HEALING_GOALS_CONFIG} is a sublist of {@link AnalyzerConfig#GOALS_CONFIG}.</li>
+   *   <li>{@link AnalyzerConfig#DEFAULT_GOALS_CONFIG} is a subset of {@link AnalyzerConfig#GOALS_CONFIG}.</li>
+   *   <li>{@link AnalyzerConfig#HARD_GOALS_CONFIG} is a subset of {@link AnalyzerConfig#DEFAULT_GOALS_CONFIG}.</li>
```

Comment:
Shouldn't we check `HARD_GOALS_CONFIG` and `SELF_HEALING_GOALS_CONFIG` are subset of `GOALS_CONFIG` instead of `DEFAULT_GOALS_CONFIG` here? 
I though `DEFAULT_GOALS_CONFIG` is only used when there user run operation without provide goal list, `GOALS_CONFIG` is a set of all supported inter-broker goals.

Reply:
1. `HARD_GOALS_CONFIG` is a subset of `DEFAULT_GOALS_CONFIG` to ensure that 
hard goal violation checks are not skipped implicitly when a user calls the `rebalance` endpoint with no arguments -- i.e. please see the relevant [sanity check](https://github.com/linkedin/cruise-control/blob/b386141146dfe4cc013b8233c19434e09b50027c/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControlUtils.java#L180).
2. `SELF_HEALING_GOALS_CONFIG` is a subset of `DEFAULT_GOALS_CONFIG` to minimize the conflicts between proposals generated via manually triggered default rebalance action and self-healing.